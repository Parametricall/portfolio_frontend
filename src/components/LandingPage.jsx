import React from 'react';
import {
    Button, Card, CardDeck, Container,
} from 'react-bootstrap';

import { LinkContainer } from 'react-router-bootstrap';
import food from '../images/food.png';
import chatBot from '../images/chat_bot.png';
import phase10 from '../images/phase_10.png';

function LandingPage() {
    return (
        <Container>
            <CardDeck className="mt-5">
                <Card>
                    {/* <Card.Img variant="top" src={food} /> */}
                    <Card.Body>
                        <Card.Title>Cookbook</Card.Title>
                        <Card.Text>
                            A simple cookbook to help keep track of your favourite recipes.
                        </Card.Text>
                    </Card.Body>
                    <Card.Footer>
                        <LinkContainer to="/cookbook">
                            <Button variant="primary">Open App</Button>
                        </LinkContainer>
                    </Card.Footer>
                </Card>
                {/* // <Card> */}
                {/* //   <Card.Img variant="top" src={chatBot}/> */}
                {/*  <Card.Body> */}
                {/*    <Card.Title>Chat System (Coming Soon)</Card.Title> */}
                {/*    <Card.Text> */}
                {/*      An experimental chat system to talk to other active users. */}
                {/*    </Card.Text> */}
                {/*  </Card.Body> */}
                {/*  <Card.Footer> */}
                {/*    <Button variant="primary">Go somewhere</Button> */}
                {/*  </Card.Footer> */}
                {/* </Card> */}
                {/* <Card> */}
                {/*  <Card.Img variant="top" src={phase10}/> */}
                {/*  <Card.Body> */}
                {/*    <Card.Title>Phase 10 (Coming in 2021)</Card.Title> */}
                {/*    <Card.Text> */}
                {/*      My attempt at making a version of the Phase 10 card game, which will suport multiple players. */}
                {/*    </Card.Text> */}
                {/*  </Card.Body> */}
                {/*  <Card.Footer> */}
                {/*    <Button variant="primary">Go somewhere</Button> */}
                {/*  </Card.Footer> */}
                {/* </Card> */}
            </CardDeck>
            {/* // <CardDeck className="mt-5"> */}
            {/* //   <Card> */}
            {/* //     <Card.Body> */}
            {/* //       <Card.Title>Snake (Coming Soon)</Card.Title> */}
            {/* //       <Card.Text> */}
            {/* //         A simple game of classic Snake. */}
            {/* //       </Card.Text> */}
            {/* //     </Card.Body> */}
            {/* //     <Card.Footer> */}
            {/* //       <LinkContainer to='/snake_game'> */}
            {/* //         <Button variant="primary">Start Game</Button> */}
            {/* //       </LinkContainer> */}
            {/* // */}
            {/* //     </Card.Footer> */}
            {/* //   </Card> */}
            {/* //   <Card> */}
            {/* //     <Card.Body> */}
            {/* //       <Card.Title>Todo List (Coming in 2021)</Card.Title> */}
            {/* //       <Card.Text> */}
            {/* //         A basic todo list app to help keep track of important tasks. */}
            {/* //       </Card.Text> */}
            {/* //     </Card.Body> */}
            {/* //     <Card.Footer> */}
            {/* //       <Button variant="primary">Go somewhere</Button> */}
            {/* //     </Card.Footer> */}
            {/* //   </Card> */}
            {/* //   <Card> */}
            {/* //     <Card.Body> */}
            {/* //       <Card.Title>Web Scraper (Coming in 2021)</Card.Title> */}
            {/* //       <Card.Text> */}
            {/* //         An app that will scrap the internet for certain information. Maybe cheapest airplane prices, or */}
            {/* //         an alarm for when a certain item goes on sale. */}
            {/*      </Card.Text> */}
            {/*    </Card.Body> */}
            {/*    <Card.Footer> */}
            {/*      <Button variant="primary">Go somewhere</Button> */}
            {/*    </Card.Footer> */}
            {/*  </Card> */}
            {/* </CardDeck> */}
        </Container>
    );
}

export default LandingPage;
