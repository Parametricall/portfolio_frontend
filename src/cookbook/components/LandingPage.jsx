import React, { useCallback, useEffect, useState } from "react";

import "../style.css";
import { Link as RouterLink, useRouteMatch } from "react-router-dom";
import { fetchData, fetchJsonData } from "../../utilities";
import { DESTROY_RECIPE_URL, GET_RECIPES_URL } from "../../constants";
import { connect } from "react-redux";
import { setUserAuthenticated } from "../../reduxStore/actions";
import {
  Button,
  Container,
  IconButton,
  List,
  ListItem,
  ListItemSecondaryAction,
  ListItemText,
  makeStyles,
  Snackbar,
} from "@material-ui/core";
import DeleteIcon from "@material-ui/icons/Delete";
import Alert from "@material-ui/lab/Alert";

const useStyles = makeStyles((theme) => ({
  root: {
    "& > *": {
      margin: theme.spacing(1),
    },
  },
  list: {
    // width: "100%",
    // maxWidth: 360,
    // backgroundColor: theme.palette.background.paper,
  },
}));

function LandingPage() {
  const classes = useStyles();
  let { url } = useRouteMatch();

  const [recipes, setRecipes] = useState([]);
  const [showSnackbar, setShowSnackbar] = useState(false);
  const [snackbarText, setSnackbarText] = useState(null);

  const fetchAllRecipes = useCallback(async () => {
    const recipes = await fetchJsonData(GET_RECIPES_URL, "GET");
    if (recipes) setRecipes(recipes);
    else {
      setRecipes([]);
      setSnackbarText("Failed to load recipes!");
      setShowSnackbar(true);
    }
  }, []);

  const handleDeleteRecipe = async (e, id) => {
    const deleteResponse = await fetchData(
      `${DESTROY_RECIPE_URL}${id}/`,
      "DELETE"
    );

    if (deleteResponse.ok) await fetchAllRecipes();
    else {
      setSnackbarText("Failed to delete recipe!");
      setShowSnackbar(true);
    }
  };

  // turn snackbar alerts into a generalised hook
  const handleSnackbarClose = () => {
    setSnackbarText(null);
    setShowSnackbar(false);
  };

  useEffect(() => {
    fetchAllRecipes().then();
  }, [fetchAllRecipes]);

  return (
    <Container maxWidth="md">
      <div className={classes.root}>
        <Button
          color="primary"
          variant="outlined"
          component={RouterLink}
          to={`${url}/create`}
        >
          Create Recipe
        </Button>
        <div className={classes.list}>
          <List>
            {recipes.map((recipe) => (
              <ListItem
                component={RouterLink}
                to={`${url}/${recipe.id}`}
                button
                key={recipe.id}
                divider
              >
                <ListItemText primary={recipe.name} />
                <ListItemSecondaryAction>
                  <IconButton
                    onClick={(e) => handleDeleteRecipe(e, recipe.id)}
                    edge="end"
                    aria-label="delete"
                  >
                    <DeleteIcon />
                  </IconButton>
                </ListItemSecondaryAction>
              </ListItem>
            ))}
          </List>
        </div>
        <Snackbar
          open={showSnackbar}
          autoHideDuration={3000}
          onClose={handleSnackbarClose}
        >
          <Alert variant="filled" severity="error">
            {snackbarText}
          </Alert>
        </Snackbar>
      </div>
    </Container>
  );
}

export default connect(null, { setUserAuthenticated })(LandingPage);
