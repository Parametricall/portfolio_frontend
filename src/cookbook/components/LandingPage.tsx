import React, { useCallback, useEffect, useState } from 'react';

import '../style.css';
import { Link as RouterLink, useRouteMatch } from 'react-router-dom';
import { connect } from 'react-redux';
import {
    Button,
    Container,
    IconButton,
    List,
    ListItem,
    ListItemSecondaryAction,
    ListItemText,
    makeStyles,
    Snackbar,
} from '@material-ui/core';
import DeleteIcon from '@material-ui/icons/Delete';
import Alert from '@material-ui/lab/Alert';
import { setUserAuthenticated } from '../../reduxStore/actions';
import { DESTROY_RECIPE_URL, GET_RECIPES_URL } from '../../constants';
import { fetchData, fetchJsonData } from '../../utilities';
// import { adminRole, guest_role, userInGroups } from '../../components/Authorization';
import UserPermissions from '../../components/UserPermissions';

const useStyles = makeStyles((theme) => ({
    root: {
        '& > *': {
            margin: theme.spacing(1),
        },
    },
    list: {
    // width: "100%",
    // maxWidth: 360,
    // backgroundColor: theme.palette.background.paper,
    },
}));

// @ts-ignore
function LandingPage() {
    const classes = useStyles();
    const { url } = useRouteMatch();

    // const user = useSelector((state: any) => state.user);

    // const adminUser = userInGroups(user, [adminRole, guest_role]);

    const [recipes, setRecipes] = useState([]);
    const [showSnackbar, setShowSnackbar] = useState<any>(false);
    const [snackbarText, setSnackbarText] = useState<any>(null);

    const fetchAllRecipes = useCallback(async () => {
        const allRecipes = await fetchJsonData(GET_RECIPES_URL, 'GET');
        if (allRecipes) setRecipes(allRecipes);
        else {
            setRecipes([]);
            setSnackbarText('Failed to load recipes!');
            setShowSnackbar(true);
        }
    }, []);

    const handleDeleteRecipe = async (e: any, id: any) => {
        const deleteResponse: any = await fetchData(
            `${DESTROY_RECIPE_URL}${id}/`,
            'DELETE',
        );

        if (deleteResponse.ok) await fetchAllRecipes();
        else {
            setSnackbarText('Failed to delete recipe!');
            setShowSnackbar(true);
        }
    };

    // turn snackbar alerts into a generalised hook
    const handleSnackbarClose = () => {
        setSnackbarText(null);
        setShowSnackbar(false);
    };

    useEffect(() => {
        fetchAllRecipes().then();
    }, [fetchAllRecipes]);

    return (
        <Container maxWidth="md">
            <div className={classes.root}>
                <UserPermissions permissions={['cookbook.add_recipe']}>
                    <Button
                        color="primary"
                        variant="outlined"
                        component={RouterLink}
                        to={`${url}/create`}
                    >
                        Create Recipe
                    </Button>
                </UserPermissions>

                <div className={classes.list}>
                    <List>
                        {recipes.map((recipe: any) => (
                            <ListItem
                                component={RouterLink}
                                to={`${url}/${recipe.id}`}
                                button
                                key={recipe.id}
                                divider
                            >
                                <ListItemText primary={recipe.name} />
                                <ListItemSecondaryAction>
                                    <UserPermissions permissions={['cookbook.delete_recipe']}>
                                        <IconButton
                                            onClick={(e) => handleDeleteRecipe(e, recipe.id)}
                                            edge="end"
                                            aria-label="delete"
                                        >
                                            <DeleteIcon />
                                        </IconButton>
                                    </UserPermissions>
                                </ListItemSecondaryAction>
                            </ListItem>
                        ))}
                    </List>
                </div>
                <Snackbar
                    open={showSnackbar}
                    autoHideDuration={3000}
                    onClose={handleSnackbarClose}
                >
                    <Alert variant="filled" severity="error">
                        {snackbarText}
                    </Alert>
                </Snackbar>
            </div>
        </Container>
    );
}

export default connect(null, { setUserAuthenticated })(LandingPage);
