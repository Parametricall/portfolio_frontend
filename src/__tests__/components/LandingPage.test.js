import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import LandingPage from '../../components/LandingPage';


describe('Landing page', () => {
  const wrapper = shallow(<MemoryRouter><LandingPage/></MemoryRouter>);
  const landingPage = wrapper.find(LandingPage).dive();

  it('renders correctly', () => {
    expect(landingPage).toMatchSnapshot();
  });
});