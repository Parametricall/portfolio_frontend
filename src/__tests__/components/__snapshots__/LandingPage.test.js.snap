// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page renders correctly 1`] = `
<Container
  fluid={false}
>
  <CardDeck
    className="mt-5"
  >
    <Card
      body={false}
    >
      <CardImg
        src="food.png"
        variant="top"
      />
      <CardBody>
        <CardTitle>
          Cookbook (Coming Soon)
        </CardTitle>
        <CardText>
          A simple cookbook to help keep track of your favourite recipes.
        </CardText>
      </CardBody>
      <CardFooter>
        <withRouter(LinkContainer)
          to="/cookbook"
        >
          <Button
            active={false}
            disabled={false}
            variant="primary"
          >
            Open App
          </Button>
        </withRouter(LinkContainer)>
      </CardFooter>
    </Card>
    <Card
      body={false}
    >
      <CardImg
        src="chat_bot.png"
        variant="top"
      />
      <CardBody>
        <CardTitle>
          Chat System (Coming Soon)
        </CardTitle>
        <CardText>
          An experimental chat system to talk to other active users.
        </CardText>
      </CardBody>
      <CardFooter>
        <Button
          active={false}
          disabled={false}
          variant="primary"
        >
          Go somewhere
        </Button>
      </CardFooter>
    </Card>
    <Card
      body={false}
    >
      <CardImg
        src="phase_10.png"
        variant="top"
      />
      <CardBody>
        <CardTitle>
          Phase 10 (Coming in 2021)
        </CardTitle>
        <CardText>
          My attempt at making a version of the Phase 10 card game, which will suport multiple players.
        </CardText>
      </CardBody>
      <CardFooter>
        <Button
          active={false}
          disabled={false}
          variant="primary"
        >
          Go somewhere
        </Button>
      </CardFooter>
    </Card>
  </CardDeck>
  <CardDeck
    className="mt-5"
  >
    <Card
      body={false}
    >
      <CardBody>
        <CardTitle>
          Todo List (Coming in 2021)
        </CardTitle>
        <CardText>
          A basic todo list app to help keep track of important tasks.
        </CardText>
      </CardBody>
      <CardFooter>
        <Button
          active={false}
          disabled={false}
          variant="primary"
        >
          Go somewhere
        </Button>
      </CardFooter>
    </Card>
    <Card
      body={false}
    >
      <CardBody>
        <CardTitle>
          Web Scraper (Coming in 2021)
        </CardTitle>
        <CardText>
          An app that will scrap the internet for certain information. Maybe cheapest airplane prices, or an alarm for when a certain item goes on sale.
        </CardText>
      </CardBody>
      <CardFooter>
        <Button
          active={false}
          disabled={false}
          variant="primary"
        >
          Go somewhere
        </Button>
      </CardFooter>
    </Card>
  </CardDeck>
</Container>
`;
